src/sirocco/workgraph.py

```diff
@@ -168,23 +190,42 @@ def _add_tasks(self):
         for task in self._core_workflow.tasks:
             self._link_wait_on_to_task(task)
 
+        # FIXME: Nothing ensures that the input data nodes are all created
+        #        before linking them to the task. It currently depends on the
+        #        order in which tasks were specified in the config file which
+        #        should have no influence here.
         for task in self._core_workflow.tasks:
             for output in task.outputs:
                 self._link_output_nodes_to_task(task, output)
-            for input_, _ in task.inputs:
+            for input_ in task.input_data_nodes():
                 self._link_input_nodes_to_task(task, input_)
-            self._link_arguments_to_task(task)
+            self._link_inputs_to_ports(task)
 
     def _create_task_node(self, task: core.Task):
-        label = AiidaWorkGraph.get_aiida_label_from_graph_item(task)
+        label = self.get_aiida_label_from_graph_item(task)
         if isinstance(task, core.ShellTask):
-            command_path = Path(task.command)
-            command_full_path = task.command if command_path.is_absolute() else task.config_rootdir / command_path
-            command = str(command_full_path)
+            # Split command line between command and arguments (this is required by aiida internals)
+            cmd, _ = self.split_cmd_arg(task.command)
+            cmd_path = Path(cmd)
+            # FIXME: task.config_rootdir shouldn't be used here. The final behavior of Sirocco should be:
+            #        1- src is copied to the task working directory on the computer
+            #        2- If the command is gven with a relative path, it can target any executable in $PATH, e.g.:
+            #           - relative path to the task working directory (./my_script.sh)
+            #           - something added to $PATH through environment activation (cdo)
+            #        So the full path to the command can only be resolved at runtime.
+            #        See issue https://github.com/C2SM/Sirocco/issues/127
+            if cmd_path.is_absolute():
+                command = str(cmd_path)
+            else:
+                if task.src is None:
+                    msg = "src must be specified when command path is relative"
+                    raise ValueError(msg)
+                command = str((task.config_rootdir / task.src).parent / cmd_path)
 
             # metadata
             metadata: dict[str, Any] = {}
             ## Source file
+            # FIXME: Same as above
```

Comment:
I think it is good to add this everywhere, sometimes code is moved around and then "above" is not so clear anymore
```suggestion
            # FIXME: See issue  https://github.com/C2SM/Sirocco/issues/127
```